src/webapi/element.rs

```diff
@@ -225,6 +226,40 @@ pub trait IElement: INode + IParentNode + IChildNode {
     fn insert_html_after( &self, html: &str ) -> Result<(), InsertAdjacentError> {
         self.insert_adjacent_html(InsertPosition::AfterEnd, html)
     }
+
+    /// The slot property of the Element interface returns the name of the shadow DOM
+    /// slot the element is inserted in.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/slot)
+    // https://dom.spec.whatwg.org/#dom-element-slot
+    fn slot( &self ) -> String {
+        js!(
+            return @{self.as_ref()}.slot;
+        ).try_into().unwrap()
+    }
+
+    /// Attach a shadow DOM tree to the specified element and returns a reference to its `ShadowRoot`.
+    /// It returns a shadow root if successfully attached or `None` if the element cannot be attached.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow)
+    // https://dom.spec.whatwg.org/#dom-element-attachshadow
+    fn attach_shadow( &self, mode: ShadowRootMode ) -> Option<ShadowRoot> {
```

Comment:
This can throw hence it should return a `Result`. Search for `js_try` and `error_enum_boilerplate!` for examples on how to fix this.